charts/promitor/templates/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "promitor.fullname" . }}
+  labels:
+    app: {{ template "promitor.name" . }}
+    chart: {{ template "promitor.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: {{ template "promitor.name" . }}
+      release: {{ .Release.Name }}
+      type: {{ .Values.service.selectorType }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "promitor.name" . }}
+        release: {{ .Release.Name }}
+        type: {{ .Values.service.selectorType }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.service.targetPort }}
+          env:
+          - name: PROMITOR_AUTH_APPID
+            valueFrom:
+              secretKeyRef:
+                  name: {{ template "promitor.fullname" . }}
+                  key: azure-app-id
+          - name: PROMITOR_AUTH_APPKEY
+            valueFrom:
+              secretKeyRef:
+                  name: {{ template "promitor.fullname" . }}
+                  key: azure-app-key
```

Comment:
What if they want to bring their own secret? Should we make it configurable? Does that make sense?